src/Projection/MySQLEventStoreProjectionTrait.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * This file is part of the prooph/pdo-event-store.
+ * (c) 2016-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\PDO\Projection;
+
+use PDO;
+use Prooph\EventStore\PDO\Exception\RuntimeException;
+
+trait MySQLEventStoreProjectionTrait
+{
+    /**
+     * @throws RuntimeException
+     */
+    protected function acquireLock(): void
+    {
+        $statement = $this->connection->prepare("SELECT GET_LOCK('$this->name', 1) as 'lock';");
```

Comment:
Ok but don't forget to expose the timeout value in the config of the package so the user can configure it. I would actually be in favor of letting the user chose how he really wants to manage his locks by exposing a locking interface. Another solution would be to make your locked column a datetime meaning until when the projection is locked (instead of storing a random int).